### PR TITLE
Replace InvenioRecordIndexer with InspireRecordIndexer

### PR DESCRIPTION
**Pull request by pazembrz**
updated tests. Fixing issue with missing citation_count on ES

InvenioRecordIndexer assumes that all records are (Invenio) Records, so when record is passed to ES it cannot check its own citations count as the method for it is a part of InspireRecord
Thanks to this change ES will be now populated with proper data about citations
INSPIR-1126

## Checklist:

- [x] I have all the information that I need (if not, move to `RFC` and look for it).
- [x] I linked the related issue(s) in the corresponding commit logs.
- [x] I wrote [good commit log messages](https://github.com/torvalds/subsurface-for-dirk/blob/5f15ad5a86ada3c5e574041a5f9d85235322dabb/README#L92-L119).
- [x] My code follows the code style of this project.
- [ ] I've added any new docs if API/utils methods were added.
- [x] I have updated the existing documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: inspirehep/modules/records/indexer.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+from invenio_indexer.api import RecordIndexer
+
+from inspirehep.modules.records.api import InspireRecord
+
+
+class InspireRecordIndexer(RecordIndexer):
+    def index_by_id(self, record_uuid):
+        """Index a record by record identifier.
+        :param record_uuid: Record identifier.
```
Comment (ammirate) [inline inspirehep/modules/records/indexer.py]: Currently we are using Napoleon style for docstrings :point_right:  https://sphinxcontrib-napoleon.readthedocs.io/en/latest/

File: inspirehep/modules/records/indexer.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        return self.index(InspireRecord.get_record(record_uuid))
+
+    def delete_by_id(self, record_uuid):
+        """Delete record from index by record identifier."""
+        self.delete(InspireRecord.get_record(record_uuid))
+
+    def _delete_action(self, payload):
+        """Bulk delete action.
+        :param payload: Decoded message body.
+        :returns: Dictionary defining an Elasticsearch bulk 'delete' action.
```
Comment (ammirate) [inline inspirehep/modules/records/indexer.py]: same here

File: tests/integration/records/test_records_serializers_json.py [hunk 3]
```diff
@@ -148,21 +148,58 @@ def test_zero_citations_in_vnd_plus_inspire_record_ui_json(isolated_api_client):
                                        headers={'Accept': 'application/vnd+inspire.record.ui+json'})
     assert response.status_code == 200
     result = json.loads(response.get_data(as_text=True))
-    assert result['metadata']['citations_count'] == 0
+    assert result['metadata']['citation_count'] == 0
 
 
 def test_non_zero_citation_count_in_vnd_plus_inspire_record_ui_json(isolated_api_client):
+    from invenio_db import db
```
Comment (ammirate) [inline tests/integration/records/test_records_serializers_json.py]: Ops, `import` in the test :)

File: tests/integration/records/test_records_serializers_json.py [hunk 4]
```diff
@@ -148,21 +148,58 @@ def test_zero_citations_in_vnd_plus_inspire_record_ui_json(isolated_api_client):
...
+
+        # # Add second citation
+        # ref = {'control_number': 12341, 'references': [{'record': {'$ref': record._get_ref()}}]}
+        # TestRecordMetadata.create_from_kwargs(json=ref)
+        #
+        # response = isolated_api_client.get(url,
+        #                                    headers={'Accept': 'application/vnd+inspire.record.ui+json'})
+        # assert response.status_code == 200
+        # result = json.loads(response.get_data(as_text=True))
+        # assert result['metadata']['citation_count'] == 2
```
Comment (ammirate) [inline tests/integration/records/test_records_serializers_json.py]: Leftovers? :)

File: tests/integration/records/test_records_serializers_json.py [hunk 5]
```diff
@@ -148,21 +148,58 @@ def test_zero_citations_in_vnd_plus_inspire_record_ui_json(isolated_api_client):
...
+        1234: 1,
+        12345: 0
+    }
     record_json = {
-        'control_number': 123,
+        'control_number': 1234,
     }
     record = TestRecordMetadata.create_from_kwargs(json=record_json).inspire_record
-    url = '/literature/123'
+    url = '/literature/?page=1&size=25&q=control_number%3A1234'
```
Comment (ammirate) [inline tests/integration/records/test_records_serializers_json.py]: you can use `/record/1234`. There is a proxy which gets by itself the collection the record belongs to and redirect to the correct page ;)
Try for example https://labs.inspirehep.net/record/1689220.

File: tests/integration/records/test_records_serializers_json.py (hunk 4 above)
Comment (pazembrz) [reply tests/integration/records/test_records_serializers_json.py]: Not really, but may be better if I will remove them.

File: tests/integration/records/test_records_serializers_json.py (hunk 5 above)
Comment (pazembrz) [reply tests/integration/records/test_records_serializers_json.py]: But here we want the results from ES to check if it has citation_count inside. If we will use record/1234 the data would be pulled from DB not ES

File: tests/integration/records/test_records_serializers_json.py (hunk 3 above)
Comment (pazembrz) [reply tests/integration/records/test_records_serializers_json.py]: This is leftover from 'debugging' ;) - will be removed

File: tests/integration/records/test_records_serializers_json.py (hunk 5 above)
Comment (ammirate) [reply tests/integration/records/test_records_serializers_json.py]: Oh, I see your point.
Just for you to know, there also `get_es_record` :point_right: https://github.com/inspirehep/inspire-next/blob/master/inspirehep/utils/record_getter.py#L68
Choose the one you prefer :+1:

**Comment by ammirate** [comment]
ci test please

**Comment by ammirate** [comment]
ci test please

**Comment by pazembrz** [comment]
ci test please

**Comment by pazembrz** [comment]
ci test please

**Comment by pazembrz** [comment]
ci test please

**Comment by pazembrz** [comment]
ci test please

**Comment by pazembrz** [comment]
ci merge please

**Comment by pazembrz** [comment]
ci test please

**Comment by pazembrz** [comment]
ci merge please

**Pull request merged**